tools/benchmark.py

```diff
@@ -46,16 +45,13 @@ def write_json(filename, data):
 
 
 def import_data_from_gh_pages():
-    if os.path.exists(all_data_file):
-        return
-    try:
-        run([
-            "git", "clone", "--depth", "1", "-b", "gh-pages",
-            "https://github.com/denoland/deno.git", "gh-pages"
-        ])
-        shutil.copy(gh_pages_data_file, all_data_file)
-    except ValueError:
-        write_json(all_data_file, [])  # writes empty json data
+    if 'GITHUB_ACTION' not in os.environ:
+        return []  # empty json data for local runs
+    run([
+        "git", "clone", "--depth", "1", "-b", "gh-pages",
+        "https://github.com/denoland/deno.git", "gh-pages"
+    ])
+    return read_json(gh_pages_data_file)
```

Comment:
This logic is too complex. I would prefer we move the git clone to .github/workflows/build.yml ... I realize that will take some refactoring of how the data gets merged - and not a trivial fix.
 I'd happily land the other clean ups - how about reverting this part for now?

Reply:
So what you want is to remove all the code handling the historic data from `gh-pages` from `benchmark.py` and make it to just write the current result to a file (or even only outputting it to stdout)?

Then we could create another small script which runs only on the CI which merges the new data into the existing data of an `gh-pages` checkout. Do I understand this correctly?

I drop the `benchmark.py`for now, might look into it again tomorrow.

Reply:
Yes that’s what I was thinking... now that you write it out I worry it might be even more complex... but somehow I think having benchmarks.py clone is not properly separating concerns.